File: kubernetes-client-examples/javascript/kubernetes-watch-namespace.js
const k8s = require('@kubernetes/client-node');

const kubeconfig = new k8s.KubeConfig();
kubeconfig.loadFromDefault();

// Get namespace from current context
let namespace = 'default';
const currentContext = kubeconfig.getCurrentContext();
const contextObj = kubeconfig.contexts.find(ctx => ctx.name === currentContext);
if (contextObj && contextObj.namespace) {
    namespace = contextObj.namespace;
}

const main = async () => {
    try {
        const watch = new k8s.Watch(kubeconfig);

        const req = await watch.watch(
            `/api/v1/namespaces/${namespace}/pods`,
            {},
            (type, pod) => {
                console.log(`Event ${type} ${pod.metadata.name}`);
            },
            (err) => {
                if (err) {
                    console.error('Error watching Pods:', err);
                }
            }
        );

        await sleep(10000);

        // Abort the watch
        req.abort();
    } catch (err) {
        console.error(err);
    }
};

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms || 10000));
}

main();

File: kubernetes-client-examples/python/client-samples/kubernetes-watch-namespace.py
from kubernetes import client, config, watch

# Load the kubeconfig file
config.load_kube_config()

# Get the current context and namespace
contexts, active_context = config.list_kube_config_contexts()
namespace = active_context.get('context', {}).get('namespace', 'default')  # Default to 'default' namespace

# Initialize the API client
v1 = client.CoreV1Api()

# Set up the watch
w = watch.Watch()
count = 30

# Stream pod events only in the current context's namespace
for event in w.stream(v1.list_namespaced_pod, namespace=namespace, timeout_seconds=10):
    print("Event: %s %s" % (event['type'], event['object'].metadata.name))
    count -= 1
    if count == 0:
        w.stop()

print("Ended.")
